pkg/vault/contracts/VaultFactory.sol

```diff
@@ -64,28 +74,67 @@ contract VaultFactory is Authentication {
      * @param targetAddress Expected Vault address. The function will revert if the given salt does not deploy the
      * Vault to the target address.
      */
-    function create(bytes32 salt, address targetAddress) external authenticate {
-        if (isDisabled) {
-            revert VaultAlreadyCreated();
+    function create(
+        bytes32 salt,
+        address targetAddress,
+        bytes memory vaultCreationCode,
+        bytes memory vaultAdminCreationCode,
+        bytes memory vaultExtensionCreationCode
+    ) external authenticate {
+        if (vaultCreationCodeHash != keccak256(vaultCreationCode)) {
+            revert InvalidBytecode("Vault");
+        } else if (vaultAdminCreationCodeHash != keccak256(vaultAdminCreationCode)) {
+            revert InvalidBytecode("VaultAdmin");
+        } else if (vaultExtensionCreationCodeHash != keccak256(vaultExtensionCreationCode)) {
+            revert InvalidBytecode("VaultExtension");
         }
-        isDisabled = true;
 
         address vaultAddress = getDeploymentAddress(salt);
         if (targetAddress != vaultAddress) {
             revert VaultAddressMismatch();
         }
 
-        VaultAdmin vaultAdmin = new VaultAdmin(
-            IVault(vaultAddress),
-            _pauseWindowDuration,
-            _bufferPeriodDuration,
-            _minTradeAmount,
-            _minWrapAmount
+        VaultAdmin vaultAdmin = VaultAdmin(
+            payable(
```

Comment:
If I got right, the vault should not receive eth from any contract. Should VaultAdmin and VaultExtension be payable?

Reply:
the compiler requires it to be payable